storm/src/main/storm/mesos/MesosCommon.java

```diff
@@ -72,8 +72,15 @@ public static String getMesosComponentNameDelimiter(Map conf, TopologyDetails in
         .or(DEFAULT_MESOS_COMPONENT_NAME_DELIMITER);
   }
 
+  public static String timestampMillis() {
+    long now = System.currentTimeMillis();
+    long sec = now / 1000L;
+    long msec = now % 1000L;
+    return String.valueOf(sec) + "." + String.valueOf(msec);
+  }
+
   public static String taskId(String nodeid, int port) {
-    return nodeid + "-" + port;
+    return nodeid + "-" + port + "-" + timestampMillis();
```

Comment:
Let's file an issue to improve the String usage everywhere in this project. 
I don't want more scope creep in this commit (I'm working on adding support for `killTask()` & `shutdown()` to actually do stuff, maybe I should just keep those separate for now).
We should also switch to slf4j-api logging style, instead of the horrendous a + b + c stuff.